### PR TITLE
Feature/687 add delete team to edit team screen

### PR DESCRIPTION
**Pull request by AnnaEmson**
### What

Ticket: [Feature/687 add delete team to edit team screen](https://trello.com/c/zAdfTUzM/687-687-add-delete-team-to-edit-team-screen)
Plus a bit of clean up and update after backend changes. 

Screenshot 2022-03-15 at 16 30 31

Screenshot 2022-03-15 at 16 29 26

### How to review

Describe the steps required to test the changes.

### Who can review

Describe who worked on the changes, so that other people can review.

### CODE REVIEW

File: src/app/config/groups/thunks.js [hunk 1]
```diff
@@ -38,3 +37,84 @@ export const updateGroupRequest = (id, body) => dispatch => {
...
+                  if (error) {
+                      notifications.add({ type: "warning", message: error?.message || error.status, autoDismiss: 5000 });
+                  }
+                  console.error(error);
+              })
+        : teams
+              .getAll()
+              .then(response => {
+                  if (!response) return dispatch(actions.loadGroupsFailure());
+                  return dispatch(actions.loadGroupsSuccess(response));
```
Comment (joegasewicz) [inline src/app/config/groups/thunks.js:79]: Just a question, Will the the remote api always return an array if the response is successful? If there are no groups available will it return an empty array?

File: src/app/config/reducer.js [hunk 2]
```diff
@@ -674,6 +674,16 @@ export default function reducer(state = initialState, action) {
                 },
             };
         }
+        case groupsTypes.DELETE_GROUP_SUCCESS: {
+            return {
+                ...state,
+                groups: {
+                    ...state.groups,
+                    all: state.groups.all.filter(group => group.id !== action.id),
```
Comment (joegasewicz) [inline src/app/config/reducer.js:682]: Questions above `[src/app/config/groups/thunks.js](https://github.com/ONSdigital/florence/pull/703/files#diff-428c5e7b6cd34850a6563eb2f1bbc9c8d01673d835f8a8ac637a6d16b2280bd5)` line 79 refers to this always being an array even if empty.

**Review by joegasewicz — APPROVED** [review]
LGTM

File: src/app/config/groups/thunks.js (hunk 1 above)
Comment (AnnaEmson) [reply src/app/config/groups/thunks.js:79]: hi, I need to change that, I just copied this over.78 line is not needed

**Review by joegasewicz — APPROVED** [review]

**Pull request merged**